service/controller/resource/statuscondition/create_test.go

```diff
@@ -20,36 +19,239 @@ import (
+				unittest.GetCreatedCondition(60),
+				unittest.GetCreatingCondition(90),
+			},
+			versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+				unittest.GetVersion(60, "8.7.5"),
+			},
+			operatorVersion: "8.7.6",
+
+			expectCondition: "Updating",
+			expectVersion:   "8.7.5",
+		},
+		// Some nodes are not ready yet
+		{
+			name: "case 5",
+
+			controlPlane: nodeConfig{1, 0},
+			nodePools:    []nodeConfig{{2, 2}, {2, 0}},
+			conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+				unittest.GetUpdatingCondition(15),
+				unittest.GetCreatedCondition(60),
+				unittest.GetCreatingCondition(90),
+			},
+			versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+				unittest.GetVersion(60, "8.7.5"),
+			},
+			operatorVersion: "8.7.6",
+
+			expectCondition: "Updating",
+			expectVersion:   "8.7.5",
+		},
 		// This is the case where we simulating a cluster upgrade with condition `Updating` and we expect condition `Updated` to be set
 		{
-			name:          "case 0",
-			cluster:       unittest.DefaultCluster(),
-			ctx:           context.Background(),
-			fakek8sclient: unittest.FakeK8sClient(),
-			release:       unittest.DefaultRelease(),
+			name: "case 6",
+
+			controlPlane: nodeConfig{1, 1},
+			nodePools:    []nodeConfig{{2, 2}, {2, 2}},
+			conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+				unittest.GetUpdatingCondition(15),
+				unittest.GetCreatedCondition(60),
+				unittest.GetCreatingCondition(90),
+			},
+			versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+				unittest.GetVersion(60, "8.7.5"),
+			},
+			operatorVersion: "8.7.6",
 
 			expectCondition: "Updated",
+			expectVersion:   "8.7.6",
+		},
+		// We go HA masters
+		{
+			name: "case 7",
+
+			controlPlane: nodeConfig{3, 1},
+			nodePools:    []nodeConfig{{2, 2}, {2, 2}},
+			conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+				unittest.GetUpdatedCondition(5),
+				unittest.GetUpdatingCondition(15),
+				unittest.GetCreatedCondition(60),
+				unittest.GetCreatingCondition(90),
+			},
+			versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+				unittest.GetVersion(5, "8.7.6"),
+				unittest.GetVersion(60, "8.7.5"),
+			},
+			operatorVersion: "8.7.6",
+
+			expectCondition: "Updated",
+			expectVersion:   "8.7.6",
+		},
+		// We upgrade again
+		{
+			name: "case 8",
+
+			controlPlane: nodeConfig{3, 1},
+			nodePools:    []nodeConfig{{2, 0}, {2, 1}},
+			conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+				unittest.GetUpdatedCondition(5),
+				unittest.GetUpdatingCondition(15),
+				unittest.GetCreatedCondition(60),
+				unittest.GetCreatingCondition(90),
+			},
+			versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+				unittest.GetVersion(5, "8.7.6"),
+				unittest.GetVersion(60, "8.7.5"),
+			},
+			operatorVersion: "8.7.7",
+
+			// TODO: this should be "updating" but it does not work yet
```

Comment:
As we saw in our clusters recently, the updating condition can not appear twice.